Studies/include/AnaPyInterface.h

```diff
@@ -6,19 +6,68 @@ using LorentzVectorM = ROOT::Math::LorentzVector<ROOT::Math::PtEtaPhiM4D<float>
 using LorentzVectorE = ROOT::Math::LorentzVector<ROOT::Math::PtEtaPhiE4D<float> >;
 using LorentzVector = ROOT::Math::LorentzVector<ROOT::Math::PxPyPzE4D<float> >;
 
-LorentzVectorM getHTTp4 (const ROOT::VecOps::RVec<LorentzVectorM>& lep_p4, const ROOT::VecOps::RVec<int>& lep_genTauIndex)
+LorentzVectorM getHTTp4 (const ROOT::VecOps::RVec<LorentzVectorM>& lep_p4, const ROOT::VecOps::RVec<int>& signal_tau_idx)
 {
-    size_t n_tau = 0;
-    LorentzVector htt(0, 0, 0, 0);
+    return lep_p4.at(signal_tau_idx.at(0)) + lep_p4.at(signal_tau_idx.at(1));
+}
+
+float getHTTScalarPt (const ROOT::VecOps::RVec<LorentzVectorM>& lep_p4, const ROOT::VecOps::RVec<int>& signal_tau_idx)
+{
+    return lep_p4.at(signal_tau_idx.at(0)).Pt() + lep_p4.at(signal_tau_idx.at(1)).Pt();
+}
+
+ROOT::VecOps::RVec<int> getSignalTauIndices_Gen(const ROOT::VecOps::RVec<LorentzVectorM>& lep_p4,
+                                                const ROOT::VecOps::RVec<int>& lep_genTauIndex)
+{
+    ROOT::VecOps::RVec<int> indexes;
+    int n_tau = 0;
     for(size_t n = 0; n < lep_p4.size(); ++n) {
         if(lep_genTauIndex.at(n) >= 0) {
-            htt += lep_p4.at(n);
+            indexes.push_back(n);
             n_tau++;
         }
     }
     if(n_tau != 2)
         throw std::runtime_error("too few taus");
-    return LorentzVectorM(htt);
+    return indexes;
+}
+
+ROOT::VecOps::RVec<int> getSignalTauIndicesDeep_Tau(const ROOT::VecOps::RVec<LorentzVectorM>& lep_p4,
+                                                    const ROOT::VecOps::RVec<float>& byDeepTau2017v2p1VSjet,
+                                                    const ROOT::VecOps::RVec<float>& byDeepTau2017v2p1VSeraw,
+                                                    const ROOT::VecOps::RVec<float>& byDeepTau2017v2p1VSmuraw,
+                                                    const ROOT::VecOps::RVec<int>& lep_type,
+                                                    int channelId)
+{
+    ROOT::VecOps::RVec<size_t> ordered_index;
+    ROOT::VecOps::RVec<size_t> selected_taus_indices;
+    // values taken from: https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/python/tools/runTauIdMVA.py#L658-L685
+    double  losser_wp_vs_e = 0.0630386;
+    double  losser_wp_vs_mu = 0.1058354;
+    float light_lepton_idx;
```

Comment:
`static constexpr size_t light_lepton_idx = 0;`
(as it is implemented below, will not work anyhow if idx != 0, so, let's simplify the code.)